pkg/build/api/types.go

```diff
@@ -503,12 +526,40 @@ type BuildRequest struct {
 	// From is the reference to the ImageStreamTag that triggered the build.
 	From *kapi.ObjectReference
 
+	// Binary indicates a request to build from a binary provided to the builder
+	Binary *BinaryBuildSource
+
 	// LastVersion (optional) is the LastVersion of the BuildConfig that was used
 	// to generate the build. If the BuildConfig in the generator doesn't match, a build will
 	// not be generated.
 	LastVersion *int
 }
 
+type BinaryBuildRequestOptions struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	AsFile string
+
+	// Commit is the value identifying a specific commit
+	Commit string
+
+	// Message is the description of a specific commit
+	Message string
+
+	// AuthorName of the source control user
+	AuthorName string
+
+	// AuthorEmail of the source control user
+	AuthorEmail string
+
+	// CommitterName of the source control user
+	CommitterName string
+
+	// CommitterEmail of the source control user
+	CommitterEmail string
```

Comment:
Yeah, nested structs converting to query parameters (*Options) doesn't work
and I don't have time to fix conversion.

On Thu, Oct 22, 2015 at 1:30 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/4819#discussion_r42778803>:
>
> > +	Commit string
> > +
> > +	// Message is the description of a specific commit
> > +	Message string
> > +
> > +	// AuthorName of the source control user
> > +	AuthorName string
> > +
> > +	// AuthorEmail of the source control user
> > +	AuthorEmail string
> > +
> > +	// CommitterName of the source control user
> > +	CommitterName string
> > +
> > +	// CommitterEmail of the source control user
> > +	CommitterEmail string
>
> we don't support what? nested structs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4819/files#r42778803>.
>